docs/content/0.1.0/docs/dj-concepts/nodes.rst

```diff
@@ -2,203 +2,170 @@
-                    return func.datetime(
-                        func.strftime("%Y-%m-%dT%H:%M:00", column),
-                        type_=DateTime,
-                    )
-                ...
-            ...
-
-The first thing to notice is that ``DATE_TRUNC`` **requires** a dialect, since it's not a standard function. If the dialect is in the set of dialects that support ``DATE_TRUNC`` natively we can simply translate the function to that using ``func.date_trunc``. Note that when using a custom function we should inform SQLAlchemy of the return type, using the ``type_`` argument.
-
-If the dialect doesn't support ``DATE_TRUNC`` and is part of the SQLite family we can implement the function using other functions supported by the dialect. In the code above we're translating a call like this:
+*recent_transactions_non_trial_active_customers*
 
 .. code-block:: sql
 
-    DATE_TRUNC('minute', column)
-
-To:
+    SELECT
+    amount
+    ,purchase_date
+    ,customer_id
+    ,customer_first_name
+    ,customer_last_name
+    FROM recent_transactions_active_customers
+    WHERE account_type <> 'trial'
+
+With this design, materializing *recent_transactions_active_customers* is enough to no longer require performing a join
+to get the data for both nodes. If the filter to non-trial accounts is fast, you may choose not to materialize the
+second node at all!
+
+Dimension Nodes
+---------------
+
+One of the benefits of DJ is that it can easily find all of the available dimensions that you can use to group metrics
+as well as all of the metrics that can be grouped by a set of dimensions. Defining a dimension node includes a query to generate the dimension dataset as well as a label of the
+dimension’s primary key(s).
+
+If another node includes a foreign key for an existing dimension node, you can include a reference to the dimension
+node’s primary key in the other node’s definition. Furthermore, a dimension itself can include a foreign key that
+includes a reference to another dimension node’s primary key, meaning that dimension is also available as a second-join
+dimension. This metadata is what allows DJ to understand the relationships between metrics and dimensions and allows
+abstracting away the :code:`JOIN` and :code:`GROUP BY` clauses required to bring metrics and dimensions together!
+
+Metric Nodes
+------------
 
-.. code-block:: sql
+The primary component of a request for SQL or data from a DJ server is always one or more metrics. A metric node is
+defined as a single column from another existing node as well as an aggregation expression. If the existing node has
+other columns that are connected to dimension nodes, those dimensions will be revealed by DJ as available dimensions
+with which the metric can be grouped by. Additional dimensions will also be available if first-join dimensions contain
+foreign key(s) to other dimensions.
 
-    DATETIME(STRFTIME("%Y-%m-%dT%H:%M:00", column))
+Cube Nodes
```

Comment:
I am starting to love this idea. We will need to open up the attributes on the cube nodes to account for the various approaches to grouping columns. Say:
```
group_by: {
  columns: [col1, ...],
  grouping_sets: [columns, ...],
  cube: columns
}
```
https://www.postgresql.org/docs/15/queries-table-expressions.html#QUERIES-GROUPING-SETS